### PR TITLE
Add Clubs API

### PR DESCRIPTION
**Pull request by gotlougit**
Right now, all the info about every club is fetched for prototyping purposes.

WARNING: the code may not work because there had been an undocumented schema change which will most likely require change in structs.

In particular, the way the code steps through the groups (by iterating through ```discord```) may have to be changed later on.

### CODE REVIEW

File: pkg/query/schema.sql [hunk 1]
```diff
@@ -34,3 +34,59 @@ CREATE TABLE IF NOT EXISTS course (
     outcomes    text NOT NULL,
     PRIMARY KEY (code, branch)
 );
+
```
Comment (GetPsyched) [inline pkg/query/schema.sql:37]: Remove extra \n

File: pkg/query/models.go [hunk 2]
```diff
@@ -13,7 +13,7 @@ type Course struct {
 	Title      string
 	Branch     string
 	Semester   int16
-	Credits    []int32
+	Credits    []int16
```
Comment (GetPsyched) [inline pkg/query/models.go]: Revert this back to `int32`

File: group.go [hunk 3]
```diff
@@ -0,0 +1,160 @@
+package main
```
Comment (GetPsyched) [inline group.go]: Shift `group.go` to `pkg/handlers/` with the corresponding package

File: group.go [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+type Admin struct {
+	Group_Name  string
+	Position    string
+	Roll_Number string
+}
+
+type Member struct {
+	Group_Name  string
+	Roll_Number string
+}
```
Comment (GetPsyched) [inline group.go]: Remove redundant struct types

File: group.go [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+		resp["message"] = "Error! The query went wrong somewhere!"
+		jsonResp, _ := json.Marshal(resp)
+		w.WriteHeader(http.StatusBadRequest)
+		w.Write(jsonResp)
+		return
+	}
+
+}
+
+func GetAllMemberInfo(w http.ResponseWriter, r *http.Request) {
```
Comment (GetPsyched) [inline group.go]: Update function to reflect `sqlc`'s changes

**Comment by gotlougit** [comment]
Migration to sqlc has been completed. The Club API now seems to be compliant with the other APIs offered by the backend.

Note: Further testing may be needed

File: pkg/handlers/group.go [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+
+	"github.com/gorilla/mux"
+	"golang.org/x/exp/slices"
+)
+
+func GetAllMemberInfo(db *sql.DB, w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+
+	w.Header().set("Content-Type", "application/json")
+
```
Comment (GetPsyched) [inline pkg/handlers/group.go]: Header is set in `respondJSON()`. Remove this from here

File: pkg/handlers/group.go [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+				x += 1
+			}
+		}
+
+		groups[g] = group
+
+	}
+
+	jsonResp, _ := json.Marshal(groups)
+	w.Write(jsonResp)
```
Comment (GetPsyched) [inline pkg/handlers/group.go]: Redundant code. The following line does this job already

File: pkg/handlers/group.go [hunk 8]
```diff
@@ -0,0 +1,110 @@
...
+	
+	for g:=0; g < len(discord); g++ {
+		
+		var group map[string]interface{}
+
+		group["name"] := discord[g]["name"]
+		group["alias"] := discord[g]["alias"]
+		group["branch"] := discord[g]["branch"]
+		group["kind"] := discord[g]["kind"]
+		group["description"] := discord[g]["description"]
```
Comment (GetPsyched) [inline pkg/handlers/group.go]: The `group` variable is already declared, do not use `:=`; use `=` directly instead

File: pkg/handlers/group.go [hunk 9]
```diff
@@ -0,0 +1,110 @@
...
+
+
+		var group["faculty"] []map[string]string
+		x :=
```
Comment (GetPsyched) [inline pkg/handlers/group.go]: I'm assuming you meant to use this as an index variable. Delete it. And use `i` instead given in the next comment.

File: pkg/handlers/group.go [hunk 10]
```diff
@@ -0,0 +1,110 @@
...
+		group["alias"] := discord[g]["alias"]
+		group["branch"] := discord[g]["branch"]
+		group["kind"] := discord[g]["kind"]
+		group["description"] := discord[g]["description"]
+
+
+		var group["faculty"] []map[string]string
+		x := 
+		for f:=0; f < len(faculty); f++ {
+			thisfaculty = faculty[f]
```
Comment (GetPsyched) [inline pkg/handlers/group.go]: Use better suited syntax for this kind of looping:
```go
for i, thisFaculty := range faculty {
```

**Pull request merged**